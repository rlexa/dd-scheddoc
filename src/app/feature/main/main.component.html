@let user = user$ | async;
@let isAdmin = isAdmin$ | async;

<mat-toolbar>
  <mat-toolbar-row>
    <nav>
      <a mat-button [routerLink]="RouteCalendar" routerLinkActive="active">Kalendar</a>
      @if (isAdmin) {
        <a mat-button [routerLink]="RouteAssignment" routerLinkActive="active">Zuweisung</a>
        <a mat-button [routerLink]="RouteUsers" routerLinkActive="active">Benutzer</a>
      }
    </nav>
    <span class="sd-flex-spacer"></span>
    <span class="sd-margin-left-wide sd-margin-right-wide">SchedDoc</span>
    @if (user) {
      <mat-menu #menu="matMenu">
        <button mat-menu-item disabled>
          <mat-icon>person</mat-icon>
          <span>{{ user.displayName || user.email }}</span>
        </button>
        <button mat-menu-item (click)="logout()">
          <mat-icon>logout</mat-icon>
          <span>Ausloggen</span>
        </button>
      </mat-menu>

      <button mat-icon-button [matMenuTriggerFor]="menu"><mat-icon>person</mat-icon></button>
    }
  </mat-toolbar-row>
</mat-toolbar>

<div class="sd-space-container">
  <div class="sd-space-container-content">
    <router-outlet />
  </div>
</div>
