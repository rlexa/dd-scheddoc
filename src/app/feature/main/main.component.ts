import {CommonModule} from '@angular/common';
import {ChangeDetectionStrategy, Component} from '@angular/core';
import {MatButtonModule} from '@angular/material/button';
import {MatIconModule} from '@angular/material/icon';
import {MatToolbarModule} from '@angular/material/toolbar';
import {RouterModule} from '@angular/router';

@Component({
  selector: 'app-main',
  templateUrl: './main.component.html',
  styles: [
    `
      :host {
        display: flex;
        flex-direction: column;
        position: relative;
        height: 100%;

        nav {
          .active {
            background-color: var(--mat-sys-tertiary-container);
            color: var(--mat-sys-on-tertiary-container);
          }
        }
      }
    `,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [CommonModule, MatButtonModule, MatIconModule, MatToolbarModule, RouterModule],
})
export class MainComponent {}
