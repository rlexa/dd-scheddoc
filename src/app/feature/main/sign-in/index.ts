export * from './sign-in.directive';
