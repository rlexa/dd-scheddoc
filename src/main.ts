import {provideHttpClient} from '@angular/common/http';
import {LOCALE_ID, provideAppInitializer, provideExperimentalZonelessChangeDetection} from '@angular/core';
import {bootstrapApplication} from '@angular/platform-browser';
import {provideRouter, withComponentInputBinding} from '@angular/router';
import {AppComponent} from './app/app.component';
import {initializeLocaleFactory} from './locale';
import {RouteMain} from './routing';

bootstrapApplication(AppComponent, {
  providers: [
    // ng
    provideHttpClient(),
    provideExperimentalZonelessChangeDetection(),
    // local
    {provide: LOCALE_ID, useValue: 'de-DE'},
    provideAppInitializer(initializeLocaleFactory(['de-DE'])),
    provideRouter(
      [
        {path: RouteMain, loadChildren: () => import('./app/feature/main/routes')},
        {path: '', redirectTo: RouteMain, pathMatch: 'full'},
        {path: '**', redirectTo: ''},
      ],
      withComponentInputBinding(),
    ),
  ],
}).catch((err) => console.error(err));
