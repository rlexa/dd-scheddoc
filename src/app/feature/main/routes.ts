import {Route} from '@angular/router';
import {canActivateAsAdmin} from 'src/app/shared/can-activate-as-admin';
import {RouteAssignment, RouteCalendar, RouteUsers} from 'src/routing';
import {MainComponent} from './main.component';

export default [
  {
    path: '',
    loadComponent: () => MainComponent,
    children: [
      {path: RouteAssignment, loadChildren: () => import('../assignment/routes'), canActivate: [canActivateAsAdmin]},
      {path: RouteCalendar, loadChildren: () => import('../calendar/routes')},
      {path: RouteUsers, loadChildren: () => import('../users/routes'), canActivate: [canActivateAsAdmin]},
      {path: '**', pathMatch: 'prefix', redirectTo: RouteCalendar},
    ],
  },
] as Route[];
