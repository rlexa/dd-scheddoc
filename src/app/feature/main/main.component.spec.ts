import {MockBuilder, MockedComponentFixture, MockRender} from 'ng-mocks';
import {MainComponent} from './main.component';

describe('MainComponent', () => {
  let fixture: MockedComponentFixture<MainComponent>;

  beforeEach(() => MockBuilder(MainComponent));

  beforeEach(() => {
    fixture = MockRender(MainComponent);
    fixture.detectChanges();
  });

  it('has instance', () => expect(fixture.componentInstance).toBeTruthy());

  it('renders', () => expect(fixture).toMatchSnapshot());
});
