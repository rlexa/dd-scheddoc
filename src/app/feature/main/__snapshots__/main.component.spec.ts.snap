// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent renders 1`] = `
<mock-render>
  <app-main>
    <mat-toolbar>
      <mat-toolbar-row>
        <nav>
          <a
            mat-button=""
            ng-reflect-router-link="ca"
            ng-reflect-router-link-active="active"
            routerlinkactive="active"
          >
            Kalendar
          </a>
          <a
            mat-button=""
            ng-reflect-router-link="as"
            ng-reflect-router-link-active="active"
            routerlinkactive="active"
          >
            Zuweisung
          </a>
          <a
            mat-button=""
            ng-reflect-router-link="us"
            ng-reflect-router-link-active="active"
            routerlinkactive="active"
          >
            Benutzer
          </a>
        </nav>
        <span
          class="sd-flex-spacer"
        />
        <span
          class="sd-margin-left-wide sd-margin-right-wide sd-mobile-invisible"
        >
          SchedDoc
        </span>
        <mat-menu>
          <button
            disabled=""
            mat-menu-item=""
            ng-reflect-disabled=""
          >
            <mat-icon>
              person
            </mat-icon>
            <span>
              displayName
            </span>
          </button>
          <button
            mat-menu-item=""
          >
            <mat-icon>
              logout
            </mat-icon>
            <span>
              Ausloggen
            </span>
          </button>
        </mat-menu>
        <button
          mat-icon-button=""
          ng-reflect-menu="[object Object]"
        >
          <mat-icon>
            person
          </mat-icon>
        </button>
      </mat-toolbar-row>
    </mat-toolbar>
    <div
      class="sd-space-container"
    >
      <div
        class="sd-space-container-content"
      >
        <router-outlet />
      </div>
    </div>
  </app-main>
</mock-render>
`;
